docs/tutorials/on-device-training.md

```diff
@@ -0,0 +1,1406 @@
+   private var nameClassA: String = "A" // Default class A name
+   private var samplesClassB = 0 // Number of samples for class B
+   private var nameClassB: String = "B" // Default class B name
+   private var samplesClassX = 0 // Number of samples for class X
+   private var nameClassX: String = "X" // Default class X name
+   private var samplesClassY = 0 // Number of samples for class Y
+   private var nameClassY: String = "Y" // Default class Y name
+   private val prepackedDefaultLabels: Array<String> = arrayOf("dog", "cat", "elephant", "cow") // Default labels for non custom class
+   ```
+
+   i. Launching the application
+
+      When the application launches, the `onCreate` function gets invoked. This function is responsible for setting up the session cache and the user interface handlers.
+
+      The code for this function is:
+
+   ```java
+   override fun onCreate(savedInstanceState: Bundle?) {
+      super.onCreate(savedInstanceState)
+
+      binding = ActivityMainBinding.inflate(layoutInflater)
+      setContentView(binding.root)
+
+      val trainingModelPath = copyAssetToCacheDir("training_model.onnx", "training_model.onnx")
+      val evalModelPath = copyAssetToCacheDir("eval_model.onnx", "eval_model.onnx")
+      val checkpointPath = copyFileOrDir("checkpoint")
+      val optimizerModelPath = copyAssetToCacheDir("optimizer_model.onnx", "optimizer_model.onnx")
+      ort_session = createSession(checkpointPath, trainingModelPath, evalModelPath, optimizerModelPath,
+            "$cacheDir")
+
+      val inferButton: Button = findViewById(R.id.infer_button)
+      inferButton.setOnClickListener(onInferenceButtonClickedListener)
+
+      val trainButton: Button = findViewById(R.id.train_button)
+      trainButton.setOnClickListener(onTrainButtonClickedListener)
+      trainButton.isEnabled = false
+
+      val classA: Button = findViewById(R.id.classA)
+      classA.setOnClickListener(onClassAClickedListener)
+      classA.setOnLongClickListener(onClassALongClickedListener)
+
+      val classB: Button = findViewById(R.id.classB)
+      classB.setOnClickListener(onClassBClickedListener)
+      classB.setOnLongClickListener(onClassBLongClickedListener)
+
+      val classX: Button = findViewById(R.id.classX)
+      classX.setOnClickListener(onClassXClickedListener)
+      classX.setOnLongClickListener(onClassXLongClickedListener)
+
+      val classY: Button = findViewById(R.id.classY)
+      classY.setOnClickListener(onClassYClickedListener)
+      classY.setOnLongClickListener(onClassYLongClickedListener)
+
+      binding.customClassSetting.setOnCheckedChangeListener(onCustomClassSettingChangedListener)
+
+      // Home screen
+      binding.statusMessage.text = "ORT Personalize"
+   }
+   ```
+
+   j. Custom class button handlers - We would like to use the class buttons for users to select their custom images for training. We need to add the listeners for these buttons to do this. These listeners will do exactly that:
```

Comment:
looks we almost explain the code in very details? is this needed in this doc, or they are more likely to part of code comments. I know you must already spend lots of time on them, want to know how you guys think about it.

Reply:
From the discussion we had internally, we wanted to create a very detailed code walkthrough showing how to start a project from start to finish. 
But I feel you're right. I can move all non ort related code out of this file if that is the consensus.